views/js/testRunner/actionBar/button.js

```diff
@@ -0,0 +1,558 @@
+         * Removes the button active state
+         * @returns {button}
+         */
+        clearActive : function clearActive() {
+            if (this.is('button')) {
+                this.$button.removeClass('active');
+            } else {
+                this.$button.find('.active').removeClass('active');
+            }
+
+            return this;
+        },
+
+        /**
+         * Gets the id of the selected menu entry
+         * @returns {String|null}
+         */
+        getActiveMenu : function getActiveMenu() {
+            var selected;
+            if (this.hasMenu()) {
+                selected = this.$menu.find('.selected').data('control');
+            }
+            return selected || null;
+        },
+
+        /**
+         * Sets the selected menu entry
+         * @param {String} id
+         * @returns {button}
+         */
+        setActiveMenu : function setActiveMenu(id) {
+            if (this.hasMenu()) {
+                this.clearActiveMenu();
+                this.$menu.find('[data-control="'+ id + '"]').addClass('selected');
+            }
+            return this;
+        },
+
+        /**
+         * Clears the menu selection
+         * @returns {button}
+         */
+        clearActiveMenu : function setActiveMenu() {
+            if (this.hasMenu()) {
+                this.$menu.find('.selected').removeClass('selected');
+            }
+            return this;
+        },
+
+        /**
+         * Additional setup onto the button config set
+         * @private
+         */
+        setup : function setup() {
+            // just a template method to be overloaded
+        },
+
+        /**
+         * Additional cleaning while uninstalling the button
+         * @private
+         */
+        tearDown : function tearDown() {
+            // just a template method to be overloaded
+        },
+
+        /**
+         * Additional DOM rendering
+         * @private
+         */
+        afterRender : function afterRender() {
+            // just a template method to be overloaded
+        },
+
+        /**
+         * Action called when the button is clicked
+         * @param {String} id
+         * @param {jQuery} $action
+         * @private
+         */
+        action : function action(id, $action) {
+            // just a template method to be overloaded
+        },
+
+        /**
+         * Action called when a menu item is clicked
+         * @param {String} id
+         * @param {jQuery} $menuItem
+         * @private
+         */
+        menuAction : function menuAction(id, $menuItem) {
+            // just a template method to be overloaded
+        },
+
+        /**
+         * Action called when the assessment item has been loaded
+         */
+        itemLoaded : function() {
+            // just a template method to be overloaded
+        }
+    };
+
+    /**
+     * Builds a button instance
+     * @returns {button}
+     */
+    var buttonFactory = function buttonFactory() {
```

Comment:
Could you modify the factory in order to have an API that is more _straight forward_. Instead of using it like : 

```javascript
_.assign(buttonFactory(), {
       isVisible : function(){ return true; }
});
```

It would be better to do 

```javascript
buttonFactory({
       isVisible : function(){ return true; }
});
```

`_.defaults` or `_.merge` should be sufficient